### ISSUE TITLE
Feature Request: Change space when throwing window to different space

### ISSUE THREAD

**Issue by kiryph** [ianyh/Amethyst#845]
## System

 - **Amethyst version**: 0.14
 - **macOS version**: 10.14.5

## What's the problem?
When I move a window to a different space, I would like to switch to this space as well at the same time.

## How can it be reproduced?

Use one of the following shortcuts

| Default Shortcut | Description |
|---|---|
| `mod2 + [left,right]` | Throw focused window to space left/right |
| `mod2 + [1-9,0]` | Throw focused window to space [1-9,10] |

## What applications are involved?

For fast switching to the left and right space, I use the third party program TotalSpaces2 v2.7..12 https://totalspaces.binaryage.com/ with following settings:

![Screenshot 2019-07-21 at 12 59 05](https://user-images.githubusercontent.com/371756/61590328-bd9c1b00-abb7-11e9-8452-086eb2050a77.png)

I hope that Amethyst can also switch to the left and right space immediately without any annoying animation. If this gets implemented, I would be glad that changing to the left and right space without animation could be separated and offered as own shortcuts provided by Amethyst making TotalSpaces optional

| Default Shortcut | Description |
|---|---|
| `mod1 + [left,right]` | Change to space left/right |

**Comment by kiryph** [comment]
If Amethyst cannot support switch spaces due to technical issues, I guess I will go with [yabai](https://github.com/koekeishiya/yabai) another macOS window manager which supports spaces.

**Comment by DigitalRogues** [comment]
I too heavily used the old throw focused window behavior, it was handy to switch spaces along with the window to see if that space was currently being used.

If there is anyway to get it back, that isn't reverting all your code of course, that'd be super helpful.

**Comment by ianyh** [comment]
It should be doable. I'm not sure if it will animate, though, which might make the UI a bit disorienting if you're using the fullscreen layout. It will look like nothing changed. I'll play with it a bit.

**Comment by kiryph** [comment]
@ianyh
>I'm not sure if it will animate

This is precisely what I would want. I do *NOT* want an animation. That is the reason why I have installed Total Spaces 2.

Furthermore, I have installed https://github.com/gechr/WhichSpace which shows me an indicator in the menu bar. I have also assigned different wallpapers for the different desktops to know on which desktop I am.

**Comment by ianyh** [comment]
I played with it a bunch. The private APIs for changing spaces state confuse Mission Control like crazy. I came upon a simple solution that animates. Focusing windows on another space moves to that space, so just manually focusing the window after throwing it to a space traverses to that space.

The full solution that does not require animation requires, as far as I can tell, some code injection into the Dock which I don't have the bandwidth to implement. This discussion seems good. https://github.com/koekeishiya/chunkwm/issues/127
